src/pykeen/models/predict.py

```diff
@@ -0,0 +1,413 @@
+        ],
+        columns=['head_id', 'head_label', 'score'],
+    ).sort_values('score', ascending=False)
+
+    return _postprocess_prediction_df(
+        df=rv,
+        add_novelties=add_novelties,
+        remove_known=remove_known,
+        training=model.triples_factory.mapped_triples,
+        testing=testing,
+        query_ids_key='head_id',
+        col=0,
+        other_col_ids=(relation_id, tail_id),
+    )
+
+
+def get_tail_prediction_df(
+    model: Model,
+    head_label: str,
+    relation_label: str,
+    add_novelties: bool = True,
+    remove_known: bool = False,
+    testing: Optional[torch.LongTensor] = None,
+) -> pd.DataFrame:
+    """Predict tails for the given head and relation (given by label).
+
+    :param model: A PyKEEN model
+    :param head_label: The string label for the head entity
+    :param relation_label: The string label for the relation
+    :param add_novelties: Should the dataframe include a column denoting if the ranked tail entities correspond
+     to novel triples?
+    :param remove_known: Should non-novel triples (those appearing in the training set) be shown with the results?
+     On one hand, this allows you to better assess the goodness of the predictions - you want to see that the
+     non-novel triples generally have higher scores. On the other hand, if you're doing hypothesis generation, they
+     may pose as a distraction. If this is set to True, then non-novel triples will be removed and the column
+     denoting novelty will be excluded, since all remaining triples will be novel. Defaults to false.
+    :param testing: The mapped_triples from the testing triples factory (TriplesFactory.mapped_triples)
+
+    The following example shows that after you train a model on the Nations dataset,
+    you can score all entities w.r.t a given head entity and relation.
+
+    >>> from pykeen.pipeline import pipeline
+    >>> result = pipeline(
+    ...     dataset='Nations',
+    ...     model='RotatE',
+    ... )
+    >>> df = result.model.get_tail_prediction_df('brazil', 'accusation')
+    """
+    head_id = model.triples_factory.entity_to_id[head_label]
+    relation_id = model.triples_factory.relation_to_id[relation_label]
+    batch = torch.tensor([[head_id, relation_id]], dtype=torch.long, device=model.device)
+    scores = model.predict_t(batch)
+    scores = scores[0, :].tolist()
+    rv = pd.DataFrame(
+        [
+            (entity_id, entity_label, scores[entity_id])
+            for entity_label, entity_id in model.triples_factory.entity_to_id.items()
+        ],
+        columns=['tail_id', 'tail_label', 'score'],
+    ).sort_values('score', ascending=False)
+
+    return _postprocess_prediction_df(
+        rv,
+        add_novelties=add_novelties,
+        remove_known=remove_known,
+        testing=testing,
+        training=model.triples_factory.mapped_triples,
+        query_ids_key='tail_id',
+        col=2,
+        other_col_ids=(head_id, relation_id),
+    )
+
```

Comment:
```suggestion

def get_relation_prediction_df(
     model: Model,
     head_label: str,
     tail_label: str,
     add_novelties: bool = True,
     remove_known: bool = False,
     testing: Optional[torch.LongTensor] = None,
 ) -> pd.DataFrame:
     """Predict relations for the given head and tail (given by label).

     :param model: A PyKEEN model
     :param head_label: The string label for the head entity
     :param tail_label: The string label for the tail entity
     :param add_novelties: Should the dataframe include a column denoting if the ranked relations correspond
      to novel triples?
     :param remove_known: Should non-novel triples (those appearing in the training set) be shown with the results?
      On one hand, this allows you to better assess the goodness of the predictions - you want to see that the
      non-novel triples generally have higher scores. On the other hand, if you're doing hypothesis generation, they
      may pose as a distraction. If this is set to True, then non-novel triples will be removed and the column
      denoting novelty will be excluded, since all remaining triples will be novel. Defaults to false.
     :param testing: The mapped_triples from the testing triples factory (TriplesFactory.mapped_triples)

     The following example shows that after you train a model on the Nations dataset,
     you can score all relations w.r.t a given head entity and tail entity.

     >>> from pykeen.pipeline import pipeline
     >>> result = pipeline(
     ...     dataset='Nations',
     ...     model='RotatE',
     ... )
     >>> df = result.model.get_relation_prediction_df('brazil', 'portugal')
     """
     head_id = model.triples_factory.entity_to_id[head_label]
     tail_id = model.triples_factory.entity_to_id[tail_label]
     batch = torch.tensor([[head_id, tail_id]], dtype=torch.long, device=model.device)
     scores = model.predict_r(batch)
     scores = scores[0, :].tolist()
     rv = pd.DataFrame(
         [
             (relation_id, relation_label, scores[relation_id])
             for relation_label, relation_id in model.triples_factory.relation_to_id.items()
         ],
         columns=['relation_id', 'relation_label', 'score'],
     ).sort_values('score', ascending=False)

      return _postprocess_prediction_df(
         rv,
         add_novelties=add_novelties,
         remove_known=remove_known,
         testing=testing,
         training=model.triples_factory.mapped_triples,
         query_ids_key='relation_id',
         col=1,
         other_col_ids=(head_id, tail_id),
     )

```